src/tallies/tally.cpp

```diff
@@ -250,6 +254,26 @@ Tally::Tally(pugi::xml_node node)
     }
   }
 
+  // Currently the pulse-height tally is only working in a selected type of
+  // settings Here these settings are checked again
+  bool exists = std::find(std::begin(scores_), std::end(scores_), ReactionType::PHT) !=
+                std::end(scores_);
+  if (exists && scores_.size() > 1) {
+    fatal_error("Error: The Pulse-Height Tally can currently not be used in combination with other tallys.");
+  }
+  if (exists && !settings::photon_transport) {
+      fatal_error("Error: The Pulse-Height Tally works only with photon transport True.");
+  }
```

Comment:
How difficult will it be to extend this capability to neutrons?

Reply:
I have thought about how useful this feature would be. To simulate neutron detectors, I think the absorption tally is enough. Are there other applications? Would users be interested in this? I could imagine that fission makes the whole thing more complicated. But if you like, we can work on it. Maybe in an update when we have implemented the tally for photons?

Reply:
There may be users interested in this, but thinking about it a bit, it does seem that it will be a little more complicated for a number of reasons so perhaps we should defer for now. Note that checking `settings::photon_transport` is actually not sufficient to determine that the simulation doesn't contain neutrons. `settings::photon_transport` is true when we run coupled neutron-photon simulations too. The most robust way would be to look through each external source distribution `openmc::model::external_sources` and see if it emits photons. However, I don't see a way to do that easily since we have different types of external sources (`IndependentSource`, `FileSource`, `CustomSourceWrapper`)... let me think more about how to handle that